subsys/bluetooth/mesh/beacon.c

```diff
@@ -449,6 +454,26 @@ static bool net_beacon_is_running(void)
 	       (bt_mesh_priv_beacon_get() == BT_MESH_FEATURE_ENABLED);
 }
 
+static struct k_work_delayable beacon_send_work;
+
+static void beacon_send_work_cb(struct k_work *work)
+{
+	static struct bt_mesh_subnet *sub_curr;
+	struct bt_mesh_subnet *sub_first = bt_mesh_subnet_next(NULL);
+	struct bt_mesh_subnet *sub_next;
+
+	do {
+		sub_next = bt_mesh_subnet_next(sub_curr);
+		if (sub_next == sub_first && sub_curr != NULL) {
+			sub_curr = NULL;
+			return;
+		}
+		sub_curr = sub_next;
+	} while (!net_beacon_for_subnet_send(sub_curr));
+
+	k_work_reschedule(&beacon_send_work, K_MSEC(20));
```

Comment:
Does this split beacons it time if advertiser is busy sending other local messages? If local message has network transmit state > 1, it will "block" the main adv instance for more than 20ms. If there are few such messages in the adv pool, this work will add beacons faster than advertiser is able to consume advs. So once the advertiser picks beacons, it will send them one by one without delay.

Reply:
> Maybe use the end cb for the sent beacon?

Yes, that is what I though about as well. That should work.